src/tests/system/tests/test_ldap.py

```diff
@@ -599,3 +599,391 @@ def test_ldap__reset_cached_timestamps_to_reflect_changes(client: Client, ldap:
     res = client.tools.getent.group("group1")
     assert res is not None, "Group should still exist"
     assert "user1" not in res.members, "User should be removed from group"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__nss_override_homedir(client: Client, ldap: LDAP):
+    """
+    :title: Test the effect of the "override_homedir" option
+    :description:
+        This test checks that the 'override_homedir' option in the [nss]
+        section of sssd.conf correctly overrides the home directory for
+        all users.
+    :setup:
+        1. Create three users in LDAP with different homeDirectory attributes.
+        2. Configure SSSD with 'override_homedir = /home/B'.
+        3. Start SSSD.
+    :steps:
+        1. Look up each user.
+        2. Verify that the home directory for each user is '/home/B'.
+    :expectedresults:
+        1. All users are found.
+        2. Each user's home directory is overridden to '/home/B'.
+    :customerscenario: True
+    """
+    ldap.user("user_with_homedir_A").add(home="/home/A")
+    ldap.user("user_with_homedir_B").add(home="/home/B")
+    ldap.user("user_with_empty_homedir").add(home="")
+
+    client.sssd.nss["override_homedir"] = "/home/B"
+    client.sssd.start()
+
+    for user in ["user_with_homedir_A", "user_with_homedir_B", "user_with_empty_homedir"]:
+        result = client.tools.getent.passwd(user)
+        assert result is not None, f"User '{user}' not found"
+        assert result.home == "/home/B", f"User '{user}' has incorrect home directory"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__nss_fallback_homedir(client: Client, ldap: LDAP):
```

Comment:
We don't have this coverage for fallback, but I think it should go with the other homedir tests in authentication as a generic provider test.

Reply:
moved